src/App.tsx

```diff
@@ -1,26 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
+const TOTAL_ITEMS_COUNT = 42;
+
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
+const items = getNumbers(1, TOTAL_ITEMS_COUNT)
   .map(n => `Item ${n}`);
 
+function getVisibleItems(
+  currentItems: string[],
+  start: number,
+  end: number,
+): string[] {
+  return currentItems.slice(start, end);
+}
+
 export const App: React.FC = () => {
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const startCount = ((page - 1) * perPage);
+
+  const endCount = startCount + perPage > TOTAL_ITEMS_COUNT
+    ? TOTAL_ITEMS_COUNT
+    : startCount + perPage;
+
+  const visibleItems = getVisibleItems(items, startCount, endCount);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${page} (items ${startCount + 1} - ${endCount} of ${TOTAL_ITEMS_COUNT})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
+              if (+event.target.value !== perPage) {
+                setPerPage(+event.target.value);
+                setPage(1);
+              }
```

Comment:
better to move this logic to the separate function as it will make a code cleaner